File: Algorithms/Insertion Sort.cpp
#include <iostream>
using namespace std;

void insertionSort(int arr[], int n)
{
    for (int i = 1; i < n; ++i) {
        int key = arr[i];
        int j = i - 1;

        while (j >= 0 && arr[j] > key) {
            arr[j + 1] = arr[j];
            j = j - 1;
        }
        arr[j + 1] = key;
    }
}

void printArray(int arr[], int n)
{
    for (int i = 0; i < n; ++i)
        cout << arr[i] << " ";
    cout << endl;
}

int main()
{
    int arr[] = { 13, 11, 6, 24, 0 };
    int n = sizeof(arr) / sizeof(arr[0]);

    insertionSort(arr, n);
    printArray(arr, n);
    return 0;
}

File: Algorithms/Insertion_Sort.java
public class insertion_Sort{

    public static void insertionSort(int arr[]){

        int n = arr.length;
        for(int j = 1 ; j < n ; j++){

            int key = arr[j];

            int i = j - 1 ;

            while( (i >= -1 ) && (arr[i] > key)){
                arr[i+1] = arr[i];
                i--;
            }
            arr[i+1] = key ;
        }
    }

    public static void main(String a[]){

        int[] arr1 = {9,14,3,2,11,56,34,31,87};

        System.out.println("Before Sorting is : ");

        for(int i : arr1){

            System.out.print(i + " ");
        }
        System.out.println();

        insertionSort(arr1);

        System.out.println("After Insertion Sort");

        for(int i : arr1){
            System.out.println(i + " ");
        }
    }
}
